### PR TITLE
[GH-678] when using --instance, show alias if available else show the instanceID

### LINKED ISSUE #678: /Jira issue subcommand are only showing instances with an alias set
When using `/jira issue` subcommand, the auto complete to pass in an instance at the end will only show instances that have an alias set. 
For ex: `/jira issue view dkh-123 --instance` after you select `--instance`, all connected instances should be shown. Instead it will only show instances where an alias has been saved.

- The command still work if you type out the URL
- All Issue type commands such as assign, and transition are affected
- The same thing happens if you attempt to use a Jira V2 style command ex: `/jira view ext-122 --instance` 
- The command will still use default instance as expected if none is passed

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
When showing the list of available `--instance` options, show the aliases if available, else show the `instanceID`.

I tested this with 
`/jira issue view JTP-1`
`/jira issue view jstp-1 --instance server-alias`
`/jira issue view jtp-1 --instance https://mmtest.atlassian.net`

`/jira assign jstp-1 jfrerich --instance server-alias `
/jira assign jtp-1 jfrerich --instance https://mmtest.atlassian.net 

#### Ticket Link
Fixes #678

### CODE REVIEW

**Comment by jfrerich** [comment]
@DHaussermann, I did find another issue.  Could you confirm that you see this also?  

When a user installs an instance but has not opened the `create issue` modal, the default instance does not get set. 

This means that all the `/jira` commands that need the default instance do not work.

**Review by mickmister — APPROVED** [review]

**Comment by mickmister** [comment]
> This means that all the `/jira` commands that need the default instance do not work.

@jfrerich Could we default to v2legacy or the first one we find that they are connected to?

File: server/instances.go [hunk 1]
```diff
@@ -407,6 +407,10 @@ func (p *Plugin) httpAutocompleteInstalledInstanceWithAlias(w http.ResponseWrite
 			out = append(out, model.AutocompleteListItem{
 				Item: alias,
 			})
+		} else {
```
Comment (levb) [inline server/instances.go:410]: suggest:
```go
item := instances.GetAlias(instanceID)
if item == "" {
  item = string(instanceID)
}
out = append(out, model.AutocompleteListItem{
				Item: item,
			})

**Review by levb — CHANGES_REQUESTED** [review]

File: server/instances.go (hunk 1 above)
Comment (jfrerich) [reply server/instances.go:410]: yes. much clearer

**Comment by DHaussermann** [comment]
@jfrerich there's a bug here that mat be a recent regression. Updating an existing alias now breaks the list of servers 
I'm running build `dd37472 ` (2nd commit)
note: Create and subscribe are not affected.

Steps:
- Set an alias
- Note that alias a non alias instances show fine after `--instance`
- Use the slash command to set a different alias on the same instance that already has one to update it
**Observed** 
A. Instance without Alias no longer shown
B. Instances with Aliases now show the URL not the saved alias 
Screen Shot 2020-10-01 at 3 30 44 PM

**Review by levb — APPROVED** [review]

**Comment by jfrerich** [comment]
@DHaussermann looking into it now

**Comment by jfrerich** [comment]
@DHaussermann please take another look.

Note that I also corrected the following commands to list an alias in the autocomplete (if available) and also accept the alias or instanceID as a value.

* /jira subscribe list --instance
* /jira instance alias
* /jira instance unalias
* /jira webhook

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested CRUD operations around alias for instances
- Tested display is correct with and without alias in slash command autocomplete with expected command 
- Tested display is correct with and without alias in create and subscribe modal
- Ensured UI updates correctly when aliases have changed
 LGTM! @jfrerich no issues found. 

Please advise if another round of dev review is needed before we merge this.

**Review by mickmister — APPROVED** [review]

**Pull request merged**